docs/architecture/domain.md

```diff
@@ -0,0 +1,117 @@
+## Definition Model
+
+![Definition](assets/definition-model.png)
```

Comment:
This graph is a little confusing to me. Why does one job split into multiple jobs?